perun-core/src/main/java/cz/metacentrum/perun/core/implApi/modules/pwdmgr/PasswordManagerModule.java

```diff
@@ -40,7 +43,7 @@
 
 	void changePassword(PerunSession sess, String userLogin, String newPassword) throws InvalidLoginException, PasswordStrengthException;
 
-	void validatePassword(PerunSession sess, String userLogin) throws InvalidLoginException;
+	void validatePassword(PerunSession sess, String userLogin, User user) throws InvalidLoginException, ExtSourceNotExistsException, WrongReferenceAttributeValueException, WrongAttributeValueException;
```

Comment:
I would prefer covering those new exceptions in ``InternalErrorException`` as did the original method ``validatePasswordAndSetExtSources()``, since they are not relevant to the caller and are expected to never happen in correctly configured instance. Please update the modules and method hierarchy accordingly. Thank you.

Reply:
Done